sklearn/utils/estimator_checks.py

```diff
@@ -631,6 +632,43 @@ def check_sample_weights_invariance(name, estimator_orig):
                                         % name)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_sample_weights_equivalence_sampling(name, estimator_orig):
+    # check that the estimators yield same results for
+    # over-sample dataset by indice filtering and using sample_weight
+    if (has_fit_parameter(estimator_orig, "sample_weight") and
+            not (hasattr(estimator_orig, "_pairwise")
```

Comment:
why don't we use the pairwise tool to make it pairwise? Or do we not properly support sample weights then?

Reply:
No idea, the base test was coming from the other sample_weight common test.